components/sharing-dialog/src/helpers/helpers.js

```diff
@@ -0,0 +1,249 @@
+import {
+    ACCESS_VIEW_ONLY,
+    ACCESS_NONE,
+    ACCESS_VIEW_AND_EDIT,
+    SHARE_TARGET_EXTERNAL,
+    SHARE_TARGET_PUBLIC,
+} from '../sharing-constants.js'
+
+/**
+ * Returns a function, that, as long as it continues to be invoked, will not be triggered. The
+ * function will be called after it stops being called for N milliseconds. If `immediate` is
+ * passed, trigger the function on the leading edge, instead of the trailing.
+ */
+
+export const debounce = (func, wait, immediate) => {
+    let timeout
+
+    return (...args) => {
+        const context = this
+
+        const later = () => {
+            timeout = null
+
+            if (!immediate) {
+                func.apply(context, args)
+            }
+        }
+
+        const callNow = immediate && !timeout
+
+        clearTimeout(timeout)
+        timeout = setTimeout(later, wait)
+
+        if (callNow) {
+            func.apply(context, args)
+        }
+    }
+}
+
+/**
+ * Name conversion
+ */
+
+export const nameToInitials = (name) => {
+    if (!name) {
+        return ''
+    }
+
+    const parts = name.split(' ')
+    const firstNameInitial = parts.length > 0 ? parts.shift().charAt(0) : ''
+    const lastNameInitial = parts.length > 0 ? parts.pop().charAt(0) : ''
+
+    return `${firstNameInitial}${lastNameInitial}`
+}
+
+/**
+ * Access and constant conversion
+ */
+
+export const convertAccessToConstant = (access) => {
+    if (typeof access === 'boolean') {
+        return access ? ACCESS_VIEW_ONLY : ACCESS_NONE
+    }
+
+    if (/^rw/.test(access)) {
```

Comment:
Not a change request, just a suggestion (same applies to the next condition as well):
```suggestion
    if (access.startsWith('rw')) {
```

Reply:
Yeah good idea, changed in 9658e900fc46b3a9a3c15bd21ea40b0ff33e5504